hack/install-kwok.sh

```diff
@@ -0,0 +1,102 @@
+set -euo pipefail
+
+# get the latest version
+KWOK_REPO=kubernetes-sigs/kwok
+# KWOK_LATEST_RELEASE=$(curl "https://api.github.com/repos/${KWOK_REPO}/releases/latest" | jq -r '.tag_name')
+KWOK_LATEST_RELEASE="v0.3.0"
+
+# make a base directory for multi-base kustomization
+HOME_DIR=$(mktemp -d)
+BASE=${HOME_DIR}/base
+mkdir ${BASE}
+
+# allow it to tolerate everything, but not run on nodes that kwok launches
+cat <<EOF > "${BASE}/tolerate-all.yaml"
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kwok-controller
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      tolerations:
+      - operator: "Exists"
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: kwok.x-k8s.io/node
+                operator: DoesNotExist
+EOF
+
+cat <<EOF > "${BASE}/kustomization.yaml"
+  apiVersion: kustomize.config.k8s.io/v1beta1
+  kind: Kustomization
+  images:
+  - name: registry.k8s.io/kwok/kwok
+    newTag: "${KWOK_LATEST_RELEASE}"
+  resources:
+  - "https://github.com/${KWOK_REPO}/kustomize/kwok?ref=${KWOK_LATEST_RELEASE}"
+  patchesStrategicMerge:
+  - tolerate-all.yaml
+EOF
+
+# Define 10 different kwok controllers to handle large load
+for let in partition-a partition-b partition-c partition-d partition-e partition-f partition-g partition-h partition-i partition-j
```

Comment:
Should the number of partitions be configurable here or do we expect that everyone is going to need 10 containers running here that cover these different partitions?

Reply:
Kwok says each controller can reliably handle about 1k nodes and 100k pods. With 10 partitions, this makes it 10k nodes and 1 M pods, which I think is past kubernetes scalability limitation recommendations of 5k nodes. Even if the k8s doc is outdated, it's the best we've got, and I think 10 partitions would cover most scales that I imagine users are at.

- [Kwok reference](https://github.com/kubernetes-sigs/kwok/tree/main#:~:text=Currently%2C%20KWOK%20can%20reliably%20maintain%201k%20nodes%20and%20100k%20pods%20easily.)
- [Kubernetes reference](https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md#:~:text=The%20table%20below%20is%20NOT%20exhaustive%20%2D%20more%20content%20is%20coming%20soon.)

Reply:
Yeah, it was more asking if we think that users are going to use this installation script to install kwok at a smaller scale. We are going to use this for _our_ scale testing, but I could see use-cases for installing kwok and running the kwok karpenter cloudprovider with a smaller-scale cluster. I'm fine waiting until we get that requested.